dagshub/auth/oauth.py

```diff
@@ -19,15 +21,25 @@ def oauth_flow(
     client_id = client_id or config.client_id
     state = uuid.uuid4()
     middle_man_request_id = hashlib.sha256(uuid.uuid4().bytes).hexdigest()
+    auth_link = f"{dagshub_url}/authorize?state={state}&client_id={client_id}" \
+                f"&middleman_request_id={middle_man_request_id}"
 
-    link_prompt = f"Go to {dagshub_url}/authorize?state={state}&client_id={client_id}" \
-                  f"&middleman_request_id={middle_man_request_id} to authorize the client."
-    print(link_prompt)
+    webbrowser.open(auth_link)
 
-    res = httpx.post(
-        f"{dagshub_url}/middleman",
-        data={"request_id": middle_man_request_id}, timeout=None
-    )
+    rich_console.print("[bold]:exclamation::exclamation::exclamation: AUTHORIZATION REQUIRED "
+                  ":exclamation::exclamation::exclamation:[/bold]", justify="center")
+    # Doing raw prints here because the rich syntax breaks in colab
+    # Printing them line by line, because the link has to be its own print in order for Colab parser to correctly parse
+    # the whole link
+    print("\n\nOpen the following link in your browser to authorize the client:")
+    print(auth_link)
```

Comment:
Does the rich text let you put colors /

Reply:
Rich does have colors, but I have to use regular boring prints here, otherwise colab breaks.
I can probably print the first line with rich and the auth link with regular print, let me check a bit later if colab gods would accept this